float64 cX
float64 cY
